File: benchmarks/fair/recursion/treesum.c
#include <stdio.h>
#include <stdlib.h>
static long long tsum(long long node,long long depth){
    if(depth==0) return node;
    return node + tsum(node*2,depth-1) + tsum(node*2+1,depth-1);
}
int main(void){const char*e=getenv("BENCH_N");long long n=e?atoll(e):0;if(n<=0)n=24;
    printf("%lld\n",tsum(1,n));return 0;}

File: benchmarks/fair/recursion/treesum.rs
fn tsum(node:i64,d:i64)->i64{ if d==0 {node} else {node+tsum(node*2,d-1)+tsum(node*2+1,d-1)} }
fn main(){let n:i64=std::env::var("BENCH_N").ok().and_then(|v|v.parse().ok()).filter(|&v|v>0).unwrap_or(24);println!("{}",tsum(1,n));}
